src/hooks/useLeaderboard.tsx

```diff
@@ -0,0 +1,82 @@
+import { useEffect, useState } from 'react';
+import {
+  LeaderboardType,
+  LeaderboardDataHandler,
+  AllLeaderboardData,
+  AllOrderings,
+} from '../types';
+import { useToast } from '@chakra-ui/react';
+import {
+  getGitHubLeaderboard,
+  getInternshipLeaderboard,
+  getLeetcodeLeaderboard,
+  getNewGradLeaderboard,
+} from '../services/leaderboard';
+import { assertTypeAndOrderingIntegrity } from '../utils';
+
+const getLeaderboardBasedOnType = (
+  type: LeaderboardType
+): LeaderboardDataHandler => {
+  switch (type) {
+    case LeaderboardType.LeetCode:
+      return getLeetcodeLeaderboard as LeaderboardDataHandler;
+    case LeaderboardType.GitHub:
+      return getGitHubLeaderboard as LeaderboardDataHandler;
+    case LeaderboardType.InternshipApplications:
+      return getInternshipLeaderboard as LeaderboardDataHandler;
+    case LeaderboardType.NewGradApplications:
+      return getNewGradLeaderboard as LeaderboardDataHandler;
+    default:
+      return async (_) => [];
+  }
+};
+
+export const useLeaderboard = (type: LeaderboardType, order: AllOrderings) => {
```

Comment:
might be cleaner to use generics here, e.g. `useLeaderboard<T>`

Reply:
How would this work?

Reply:
actually nvm this looks good. I would however move `getLeaderboardDataHandlerFromType` to `services/leaderboard.ts` though, but otherwise LGTM and is good to merge